src/iam/identhandler/fileidentifier.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "aos/iam/identhandler/fileidentifier.hpp"
+#include "aos/common/tools/fs.hpp"
+#include "aos/common/tools/memory.hpp"
+#include "log.hpp"
+
+namespace aos {
+namespace iam {
+namespace identhandler {
+
+/***********************************************************************************************************************
+ * Public
+ **********************************************************************************************************************/
+
+Error FileIdenentifier::Init(const Config& config, SubjectsObserverItf& subjectsObserver)
+{
+    mConfig           = config;
+    mSubjectsObserver = &subjectsObserver;
+    mSubjects.Clear();
+
+    auto err = ReadSystemId();
+    if (!err.IsNone()) {
+        return err;
+    }
+
+    err = ReadUnitModel();
+    if (!err.IsNone()) {
+        return err;
+    }
+
+    err = ReadSubjects();
+    if (!err.IsNone()) {
+        LOG_WRN() << "Can't read subjects: " << err.Message() << ". Empty subjects will be used";
+    }
+
+    return ErrorEnum::eNone;
+}
+
+RetWithError<StaticString<cSystemIDLen>> FileIdenentifier::GetSystemID()
+{
+    return mSystemId;
+}
+
+RetWithError<StaticString<cUnitModelLen>> FileIdenentifier::GetUnitModel()
+{
+    return mUnitModel;
+}
+
+Error FileIdenentifier::GetSubjects(Array<StaticString<cSubjectIDLen>>& subjects)
+{
+    if (subjects.MaxSize() < mSubjects.Size())
+    {
+        return ErrorEnum::eOutOfRange;
```

Comment:
Better to use no mem error. And wrap returning errors.

Reply:
done